proposals/partial-properties.md

```diff
@@ -0,0 +1,142 @@
+    public partial string Prop { get => field; set => field = value; }
+}
+```
+
+Because we have reserved the syntactic form with semicolon accessor bodies for the *defining declaration*, a partial property cannot be *automatically implemented*. We therefore adjust [Automatically implemented properties (§15.7.4)](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/classes.md#1574-automatically-implemented-properties) as follows:
+
+> An automatically implemented property (or auto-property for short), is a non-abstract, non-extern, **non-partial,** non-ref-valued property with semicolon-only accessor bodies.
+
+**Remarks**. It is useful for the compiler to be able to look at a single declaration in isolation and know whether it is a defining or an implementing declaration. Therefore we don't want to permit auto-properties by including two identical `partial` property declarations, for example. We don't think that the use cases for this feature involve implementing the partial property with an auto-property, but in cases where a trivial implementation is desired, we think the `field` keyword makes things simple enough.
+
+A partial property must have one *defining declaration* and one *implementing declaration*.
+
+**Remarks**. We also don't think it is useful to allow splitting the declaration across more than two parts, to allow different accessors to be implemented in different places, for example. Therefore we simply imitate the scheme established by partial methods.
+
+Similar to partial methods, the attributes in the resulting property are the combined attributes of the parts are concatenated in an unspecified order, and duplicates are not removed.
+
+Only the defining declaration of a partial property participates in lookup, similar to how only the defining declaration of a partial method participates in overload resolution.
+
+**Remarks**. In the compiler, we would expect that only the symbol for the defining declaration appears in the member list, and the symbol for the implementing part can be accessed through the defining symbol. However, some features like nullable analysis might *see through* to the implementing declaration in order to provide more useful behavior.
+
+```cs
+partial class C
+{
+    public partial string Prop { get; set; }
+    public partial string Prop { get => field; set => field = value; }
+
+    public C() // warning CS8618: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+    {
+    }
+}
+```
+
+### Matching signatures
+The property declarations must have the same type and ref kind. The property declarations must have the same set of accessors.
+
+The property declarations and their accessor declarations must have the same modifiers, though the modifiers may appear in a different order. This does not apply to the `extern` modifier, which may only appear on an *implementing declaration*.
```

Comment:
FWIW, it looks like partial methods allow some variation in `scoped` (see usage of `CheckValidScopedOverride` in `PartialMethodChecks`)
Some variation in tuple names is also allowed.

Reply:
Were you able to dig up our rules for `scoped` differences?

Reply:
Scoped is just a modifier, I am expecting the normal rules for modifier differences to apply.

Reply:
This can be observed for partial methods here

```cs
partial class C
{
    public partial Span<int> M(Span<int> x);
    public partial Span<int> M(scoped Span<int> x) => default; // The 'scoped' modifier of parameter 'x' doesn't match partial method declaration.CS8988
    
    public partial Span<int> M2(scoped Span<int> x);
    public partial Span<int> M2(Span<int> x) => default; // The 'scoped' modifier of parameter 'x' doesn't match partial method declaration.CS8988
}
```

Reply:
It looks like we allow some variations specifically for `scoped` on partial method, so the normal rule of "must have the same modifiers" doesn't seem right

Reply:
I am not aware of any permitted variations on `scoped` modifiers on partial methods. The method in the implementation which checks this is called `CheckValidScopedOverride`, which may be misleading, but it is also passing argument `allowVariance: false`. In other words, scoped-ness of parameters across partial methods must be exactly the same, otherwise an error (not a warning) occurs.

Reply:
Ah, thanks for catching that. Cool :-)